pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisConsumer.java

```diff
@@ -159,23 +159,23 @@ private KinesisRecordsBatch getResult(StreamPartitionMsgOffset startOffset, Stre
 
       return new KinesisRecordsBatch(recordList, startShardToSequenceNum.getKey(), isEndOfShard);
     } catch (IllegalStateException e) {
-      LOGGER.warn("Illegal state exception, connection is broken", e);
+      debugAndLogWarning("Illegal state exception, connection is broken", e);
       return handleException(kinesisStartCheckpoint, recordList);
     } catch (ProvisionedThroughputExceededException e) {
-      LOGGER.warn("The request rate for the stream is too high", e);
+      debugAndLogWarning("The request rate for the stream is too high", e);
       return handleException(kinesisStartCheckpoint, recordList);
     } catch (ExpiredIteratorException e) {
-      LOGGER.warn("ShardIterator expired while trying to fetch records", e);
+      debugAndLogWarning("ShardIterator expired while trying to fetch records", e);
       return handleException(kinesisStartCheckpoint, recordList);
     } catch (ResourceNotFoundException | InvalidArgumentException e) {
       // aws errors
-      LOGGER.error("Encountered AWS error while attempting to fetch records", e);
+      LOGGER.error("Encountered AWS error while attempting to fetch records: {}", e.getMessage());
```

Comment:
Should we keep the stack trace here?

Reply:
This error requires some user intervention most of the time. It is either due to wrong credentials, no streams existing with the name, in correct region or something else. Which is why it is logged as error and not a warning. Since it was an error log, I just kept the stacktrace.

Reply:
The stacktrace won't be logged anymore with the change. Is that intentional?

Reply:
My bad. I thought you were suggesting to remove the stacktrace. Fixed it now